### DISCUSSION TITLE
Suggestion regarding node-binance-us-api

### DISCUSSION THREAD

**Discussion by ndrosos** [community/community#116505] [Programming Help]
### Select Topic Area

Bug

### Body

Hello,
I am using the node-binance-us-api and there are cases when I need to create multiple clients with different binance.us API keys.
I noticed that when instantiating each client using new e.g.:
const exchangeClient = new BinanceUS({
        APIKEY: apiKey,
        APISECRET: apiSecret
      });
then although there are 3 independent instances, each instance contains the options of the last instance defined. 
e.g. 
const exchangeClient1 = new BinanceUS({
        APIKEY: apiKey1,
        APISECRET: apiSecret1
      });
const exchangeClient2 = new BinanceUS({
        APIKEY: apiKey2,
        APISECRET: apiSecret2
      });

And then try to make a trade with exchangeClient1, it holds the configuration of exchangeClient2.

Investigated the issue, and found out that this is caused by
function instanceWrapper( options = {} ) {
    if ( new.target ) return api( options );
    return new api( options );
}

So when using new, "new" is consumed by the wrapper which is creating an instance of the api function without new.

The workaround is to create the 2 instances without new in order to trick the module and actually create new instances.

However, in my opinion it is confusing. It should create a new instance in every case.

**Comment by SOHITVISH** [comment]
ignore #2

**Comment by pouya-heydarabadi** [comment]
ignore

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @ndrosos ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by queenofcorgis** [comment]
If you're asking how to use the piece of software that you found in a repository on GitHub, the best way to contact the maintainers of that software is to:

- Check the README for instructions on how to operate the software and pointers to documentation or troubleshooting info.
- Check the [SUPPORT file](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-support-resources-to-your-project), if one exists, for instructions on how to best contact the maintainers for support.
- Check the [CONTRIBUTING guide](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors), if one exists. Sometimes if there isn't a SUPPORT file, the CONTRIBUTING guide will give instructions on how to contact the maintainers for support.

All of these documents can be found, if they exist, in the repository where you found the software itself.
